frontend/pages/Parts/[handle].tsx

```diff
@@ -62,38 +58,47 @@ export const getServerSideProps: GetServerSideProps<PageProps> = async (
       };
    }
 
-   const searchContext = await createProductListSearchContext({
-      appId: ALGOLIA_APP_ID,
-      apiKey: productList.algolia.apiKey,
-      algoliaIndexName: ALGOLIA_DEFAULT_INDEX_NAME,
-      urlQuery: context.query,
-   });
-
-   if (searchContext == null) {
-      return {
-         notFound: true,
-      };
-   }
+   const protocol = context.req.headers.referer?.split('://')[0] || 'https';
```

Comment:
Why are we using the `referer` here? Why does the protocol of the `referer` affect the way we render the page?

Reply:
Also, the `referer` isn't going to be in the cache key in the upstream cache, so our response can't really use it to affect the output.

Reply:
To be honest I don't know, it's in the implementation [guide](https://www.algolia.com/doc/guides/building-search-ui/going-further/server-side-rendering/react-hooks/#with-nextjs) for server side rendering with Next.js.
What would you do? Just hard code `https` ?

Reply:
It looks like the `url` prop is used down the stack for the server-side rendering of the algolia history router (or something?). It appears as if we're just using it to get a "current url". I don't know precisely what that ends up being used for in algolia, but maybe it doesn't affect anything. I'd vote for yes, just choosing `https` always.

Given that we're doing the same bit of code for every page, I think we should perhaps move that to someplace common.

Reply:
Algolia uses it because it needs to know the url server side in order to get the facets. The thing I don't know is if it makes any difference to have `http` or `https`, but I doubt it. It's weird though that their guide contains such a specific piece of code, it seems like it has been written that way for some reason.

> Given that we're doing the same bit of code for every page, I think we should perhaps move that to someplace common.

That's right 👍  I already have a plan for this, but I didn't want to complicate this PR any further. The gist is that we should have a `templates` folder that exports the various kind of pages that we have (for now, it's just the product list page). This way we can reuse the code across different routes that all render the same template.